test/functional/testlib.bash

```diff
@@ -0,0 +1,502 @@
+    -inkey "$FUNC_DIR"/private.pem \
+    -outform DER -out "$(dirname "$manifest")"/Manifest.MoM.sig
+}
+
+# Creates a test environment with the basic directory structure needed to
+# validate the swupd client
+# Parameters:
+# - ENVIRONMENT_NAME: the name of the test environment, this should be typically the test name
+# - VERSION: the version to use for the test environment, if not specified the default is 10
+create_test_environment() { 
+
+	local env_name=$1 
+	local version=${2:-10}
+	local mom
+	validate_param "$env_name"
+	
+	# create all the files and directories needed
+	# web-dir files & dirs
+	sudo mkdir -p "$env_name"/web-dir/version/formatstaging
+	echo "$version" | sudo tee "$env_name"/web-dir/version/formatstaging/latest > /dev/null
+	sudo mkdir -p "$env_name"/web-dir/"$version"/files
+	sudo mkdir -p "$env_name"/web-dir/"$version"/staged
+	mom=$(create_manifest "$env_name"/web-dir/"$version" MoM)
+	sign_manifest "$mom"
+
+	# target-dir files & dirs
+	sudo mkdir -p "$env_name"/target-dir/usr/lib
+	echo "NAME=\"Clear Linux Software for Intel Architecture\"" | sudo tee "$env_name"/target-dir/usr/lib/os-release > /dev/null
+	echo "VERSION=1" | sudo tee -a "$env_name"/target-dir/usr/lib/os-release > /dev/null
+	echo "ID=clear-linux-os" | sudo tee -a "$env_name"/target-dir/usr/lib/os-release > /dev/null
+	echo "VERSION_ID=$version" | sudo tee -a "$env_name"/target-dir/usr/lib/os-release > /dev/null
+	echo "PRETTY_NAME=\"Clear Linux Software for Intel Architecture\"" | sudo tee -a "$env_name"/target-dir/usr/lib/os-release > /dev/null
+	echo "ANSI_COLOR=\"1;35\"" | sudo tee -a "$env_name"/target-dir/usr/lib/os-release > /dev/null
+	echo "HOME_URL=\"https://clearlinux.org\"" | sudo tee -a "$env_name"/target-dir/usr/lib/os-release > /dev/null
+	echo "SUPPORT_URL=\"https://clearlinux.org\"" | sudo tee -a "$env_name"/target-dir/usr/lib/os-release > /dev/null
+	echo "BUG_REPORT_URL=\"https://bugs.clearlinux.org/jira\"" | sudo tee -a "$env_name"/target-dir/usr/lib/os-release > /dev/null
+	sudo mkdir -p "$env_name"/target-dir/usr/share/clear/bundles
+
+	# state files & dirs
+	sudo mkdir -p "$env_name"/state
+	# TODO(castulo): add more dirs & files as neeeded by the tests
+
+	# export environment variables that are dependent of the test env
+	set_env_variables "$env_name"
+
+	# every environment needs to have at least the os-core bundle so this should be
+	# added by default to every test environment
+	create_bundle -L -n os-core -v "$version" -f /usr/bin/core "$env_name"
+
+}
+
+# Destroys a test environment
+# Parameters:
+# - ENVIRONMENT_NAME: the name of the test environment to be deleted
+destroy_test_environment() { 
+
+	local env_name=$1
+	local deletable=true
+	validate_path "$env_name"
+	# since the action to be performed is very destructive, at least
+	# make sure the directory does look like a test environment
+	for var in "state" "target-dir" "web-dir"; do
+		if [ ! -d "$env_name/$var" ]; then
+			deletable=false
```

Comment:
I would go for print the message and return here, rather than having the local deletable variable

Reply:
good point :)